### PR TITLE
Pretty printing bug fix

### LINKED ISSUE #17616: inaccurate rendering of pi**(1/E)
This claims to be version 1.5.dev; I just merged from the project master, so I hope this is current.  I didn't notice this bug among others in printing.pretty.

```
In [52]: pi**(1/E)                                                               
Out[52]: 
-1___
╲╱ π 

```
LaTeX and str not fooled:
```
In [53]: print(latex(pi**(1/E)))                                                 
\pi^{e^{-1}}

In [54]: str(pi**(1/E))                                                          
Out[54]: 'pi**exp(-1)'
```

### PR DESCRIPTION
**Pull request by 0sidharth**
#### References to other Issues or PRs

Fixes #17616 
Closes https://github.com/sympy/sympy/pull/17620
Closes https://github.com/sympy/sympy/pull/17628

#### Brief description of what is fixed or changed
Earlier, 
```
>>> pprint(pi**(1/exp(1)))
-1___
╲╱ π 
```
Now,
```
>>> pprint(pi**(1/exp(1)))
 ⎛ -1⎞
 ⎝ℯ  ⎠
π     
```

#### Release Notes

* printing
    * irrational powers are no longer printed with square root sign, they are printed as fractional powers

### CODE REVIEW

File: sympy/printing/pretty/pretty.py
Comment (oscarbenjamin) [inline sympy/printing/pretty/pretty.py:1960]: What is the `is_irrational` check for?

I don't think that the assumptions system should be used in the printing code.

Comment (0sidharth) [reply sympy/printing/pretty/pretty.py:1960]: not `is_irrational` means that symbols, rational numbers, and radicals like `EulerGamma` would be passed to the function. In my current implementation, I am guaranteeing that irrational numbers are printed without the square root, however, all other symbols go through a check, in whether they are simple (height and width 1), in which case they are printed with the square root, otherwise they are also printed as fractions.

What do you mean by the assumptions system?

Comment (oscarbenjamin) [reply sympy/printing/pretty/pretty.py:1960]: The `is_irrational` attribute is part of the assumptions system:
```python
In [50]: x = Symbol('x', irrational=True)

In [51]: x.is_irrational
Out[51]: True

In [52]: x = Symbol('x', rational=True)

In [53]: x.is_irrational
Out[53]: False
```
The printing code should be based only on structural properties of the expression rather than semantic properties implied by the assumptions system.

Comment (0sidharth) [reply sympy/printing/pretty/pretty.py:1960]: Can you suggest any way to distinguish between `EulerGamma` and `pi`? I was preferring printing `EulerGamma` over the square root and irrational numbers as fractions, but I am unable to distinguish between `EulerGamma` and `pi` without using `is_irrational`.

Comment (oscarbenjamin) [reply sympy/printing/pretty/pretty.py:1960]: Why should those print any differently?

Comment (0sidharth) [reply sympy/printing/pretty/pretty.py]: I think that was the ideology the previous PR author was going for, so I just went along with it. I have made the changes now.

**Comment by 0sidharth** [comment]
@oscarbenjamin I have made the change

File: sympy/printing/pretty/pretty.py
Comment (oscarbenjamin) [inline sympy/printing/pretty/pretty.py:1919]: This change should not be here

**Comment by 0sidharth** [comment]
@oscarbenjamin is this good to go?

**Comment by oscarbenjamin** [comment]
It looks reasonable to me although I don't know the printing code so well.

Perhaps @eric-wieser could review this?

File: sympy/printing/pretty/pretty.py
Comment (eric-wieser) [inline sympy/printing/pretty/pretty.py:1905]: I think you should add a comment here explaining what `den_p` means

File: sympy/printing/pretty/pretty.py
Comment (eric-wieser) [inline sympy/printing/pretty/pretty.py:1905]: Why does this function need `exp` at all? Shouldn't we just pass in `root`, and then compute `broot = self._print(root)` inside the function and use it instead of `den_p`?

**Review by eric-wieser — DISMISSED** [review]
I think this needs some work still. I think with my suggested refactor the fix should become "obvious"

File: sympy/printing/pretty/pretty.py
Comment (0sidharth) [reply sympy/printing/pretty/pretty.py:1905]: I don't fully understand what you are trying to say.  The function can be made to work instead of `expt` and the current if-else block, yes, and I will make that change, but what do you mean by `root` variable? Are you referring to the denominator `d`? Because the denominator needs to be passed.

Comment (eric-wieser) [reply sympy/printing/pretty/pretty.py:1905]: Yes, the denominator `d` is the degree of the root, right? So I'm suggesting passing `d` at the caller into an argument called `root`

File: sympy/printing/pretty/pretty.py
Comment (eric-wieser) [inline sympy/printing/pretty/pretty.py:1910]: Should this be `root is S.Two`?

Comment (0sidharth) [reply sympy/printing/pretty/pretty.py:1910]: I think there is no `Two` Attribute for S in SymPy. It is also giving an error regarding the same, so I believe `root == 2` has to be used

File: sympy/printing/pretty/pretty.py
Comment (eric-wieser) [inline sympy/printing/pretty/pretty.py:1922]: What changes if we use the simpler condition?
```suggestion
        # roots look bad if they are not a single line
        if rpretty.height() != 1:
            return self._print(base)**self._print(1/root)
```

File: sympy/printing/pretty/tests/test_pretty.py
Comment (eric-wieser) [inline sympy/printing/pretty/tests/test_pretty.py:5949]: A question for another reviewer - is this better or worse as:
```suggestion
    assert pretty(Pow(x, (1/pi))) == \
    'π ___\n\'
    '╲╱ x '
```
Are any comments in #6825 relevant?

Comment (0sidharth) [reply sympy/printing/pretty/tests/test_pretty.py:5949]: Maybe https://github.com/sympy/sympy/pull/17628#issuecomment-542438582 is also relevant

**Review by eric-wieser — COMMENTED** [review]
Code and tests looks reasonable, just a question of what opinion people have on when root notation should be used, as this PR does two things:

* Fixes the bug that made root printing nonsensical
* Applies opinionated changes as to when root printing should be used.

If reviewers would prefer to see
```
 1
 -
 π
x  
```
than
```
π ___
╲╱ x 
```
then this PR is fine to merge as is. It's easy to adjust the code either way.

File: sympy/printing/pretty/tests/test_pretty.py
Comment (eric-wieser) [reply sympy/printing/pretty/tests/test_pretty.py:5949]: Oh nice find. @oscarbenjamin, assuming your opinion is unchanged feel free to merge.

At some point it might make sense to check that the logic in `pretty` matches `latex`:

https://github.com/sympy/sympy/blob/92f31c3ff460bddc936b3b39c6d4d9d5ec0c8ece/sympy/printing/latex.py#L614-L630

But that's not necessary in this PR.

File: sympy/printing/pretty/pretty.py
Comment (0sidharth) [reply sympy/printing/pretty/pretty.py:1922]: Really the only change is that
```
z = Symbol("x_17")
pprint(7**(1/z))
```
changes from
```
   1 
 ───
 x₁₇
7  
```
to 
```
x₁₇___
 ╲╱ 7 
```

**Comment by 0sidharth** [comment]
ping @oscarbenjamin  @asmeurer @czgdp1807

**Comment by czgdp1807** [comment]
@eric-wieser Can we merge this PR? Please let us know if any changes are to be made. Thank you.

**Comment by 0sidharth** [comment]
@czgdp1807 as mentioned in the comment and review above, we were waiting for another reviewer's opinion on the printing of `Pow(x, 1/pi)` (in general, on the printing of irrational powers over the root sign or in the fractional form)

**Comment by oscarbenjamin** [comment]
I prefer the fractional form for irrational powers.

**Comment by czgdp1807** [comment]
> than
> 
> ```
> π ___
> ╲╱ x 
> ```
> 
> then this PR is fine to merge as is. It's easy to adjust the code either way.

This seems better to me, though fractional one is also correct. Anything else blocking this PR? If not, then I think either of the approaches will work. Let us know before making any change.

**Comment by 0sidharth** [comment]
> then this PR is fine to merge as is. It's easy to adjust the code either way.

If the fractional form works, then this PR is ready to merge as is. All requested changes have been addressed.

**Comment by czgdp1807** [comment]
Okay. Can be merged by tomorrow if no objections raised until then.

**Comment by asmeurer** [comment]
I'm late on this, but I also like this PR. I was worried that it would break `root(x, n)`, but I see that it still uses the surd notation if the exponent is a symbol.

**Comment by czgdp1807** [comment]
Yes, the results of `root(x, n)` are same in both `1.7` and `master`. Should we keep the printing outputs consistent i.e., same for both symbolic and numerical exponents?

**Pull request merged**
